### PR TITLE
Blog App: Integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by eudondian**
In this PR, we ( @Timbar09 and @eudondian  ) added integration specs for views and fixed n+1 problems.

**Changes**

- Solved the N+1 problem.
> [Before-N+1 fix image](https://github.com/Timbar09/Blog_App-Rails/assets/90119799/2f5792b6-c671-4e45-af18-26282d9ed45a)
> [After-N+1 fix image](https://github.com/Timbar09/Blog_App-Rails/assets/90119799/e31d87ef-6a50-4be7-9928-855a80e8547d)
- Used Capybara to write integration tests for the `index` and `show` views of both `users` and `posts`.

### CODE REVIEW

File: spec/integration/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    end
+
+    it 'should redirect to the user page when a username is clicked' do
+      find('.user_card', text: @john.name).click
+      expect(page).to have_current_path(user_path(@john))
+    end
+  end
+
+  describe 'show page' do
+    before(:example) do
```
Comment (TadesseDev) [inline spec/integration/users_spec.rb]: - [ ] Dear Team, I see you use a single `posts_spec` file for writing the tests of the `post_index` and `post_show` routes. Please be reminded we need to create a separate file for each route and test them accordingly. Please create separate `post_index_spec` and `post_show_spec` files for the tests of `post`.  This makes your code more modularized and easy to followup.

File: spec/integration/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    @post2 = Post.create(title: 'You can\'t see me', text: 'My time is now. It\'s the frenchise.', author: @john)
+    @post3 = Post.create(title: 'I will never give up', text: 'I will never give up. I will never surrender.',
+                         author: @john)
+    @post4 = Post.create(title: 'No one can beat me', text: 'No one can beat me. I am the best.', author: @john)
+    @post5 = Post.create(title: 'Like it or not', text: 'Like it or not, I am the best.', author: @john)
+  end
+
+  describe 'index page' do
+    before(:example) do
+      visit users_path
```
Comment (TadesseDev) [inline spec/integration/users_spec.rb]: - [ ] For the same reason I mentioned in my previous comment. Please create a separate file for the `user_index` and `user_show` routes.

File: spec/integration/users_spec.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    end
+
+    it 'should render user information' do
+      expect(page).to have_content(@cristiano.name)
+      expect(page).to have_content(@john.name)
+
+      expect(page).to have_css("img[src*='default_user.jpg']")
+
```
Comment (TadesseDev) [inline spec/integration/users_spec.rb]: - [ ] Dear Team, for the profile picture test, we need to make sure the profile picture of each user is visible. Hence we need to have a test for more than one user with a different profile picture. This better asserts our app is rendering all the profile pictures.

File: spec/integration/posts_spec.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+  describe 'show page' do
+    before(:example) do
+      visit user_post_path(@messi, @messi_post1)
+    end
+
+    it 'should render a user\'s post information' do
+      expect(page).to have_content(@messi_post1.title.capitalize)
+      expect(page).to have_content("by #{@messi.name}")
+      expect(page).to have_content('Comments: 7 | Likes: 0')
```
Comment (TadesseDev) [inline spec/integration/posts_spec.rb]: - [ ] For the `number of comments` and `number of likes` Please be reminded it is not recommended to use a  constant/typed number, instead we need to take the post and check if the number of comment of the post is displayed on the page. 

> Something like 
```Ruby
 expect(page).to have_content(@messi_post1.comments.count)
```

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Most of the functionality has been tested. ✅
 - [x]  Great job on solving the N+1 problem on the post controller (consider applying this in all the other appropriate places) ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rotimiazeez — APPROVED** [review]
Hi @eudondian & @Timbar09 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Suggested changes have been worked on 👍🏽
- Good teamwork 👍🏽
- All test cases pass ✔️
- No linter error 💯 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Timbar09** [comment]
Thank you for the feedback,  @rotimiazeez. 🙃

**Pull request merged**